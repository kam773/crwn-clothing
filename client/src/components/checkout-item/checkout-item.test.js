import React from 'react';
import { shallow } from 'enzyme';
import { CheckoutItem } from './checkout-item.component';
import {
  ImageContainer,
  RemoveButtonContainer,
  TextContainer,
  QuantityContainer,
} from './checkout-item.styles';
import renderer from 'react-test-renderer';

const mockProps = {
  cartItem: {
    name: 'name',
    imageUrl: 'url',
    price: '20',
    quantity: '2',
  },
  clearItem: jest.fn(),
  addItem: jest.fn(),
  removeItem: jest.fn(),
};

describe('create shanpshot', () => {
  it('capture snapshot of CheckoutItem', () => {
    const wrapper = renderer
      .create(<CheckoutItem {...mockProps} />)
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});

describe('shallow render react component', () => {
  it('render the DUMB component', () => {
    const wrapper = shallow(<CheckoutItem {...mockProps} />);
    expect(wrapper.length).toEqual(1);
  });
  it('render ImageContainer', () => {
    const wrapper = shallow(<CheckoutItem {...mockProps} />);
    expect(wrapper.find(ImageContainer)).toBeDefined();
  });
  it('render RemoveButtonContainer', () => {
    const wrapper = shallow(<CheckoutItem {...mockProps} />);
    expect(wrapper.find(RemoveButtonContainer)).toBeDefined();
  });
  it('render TextContainer', () => {
    const wrapper = shallow(<CheckoutItem {...mockProps} />);
    expect(wrapper.find(TextContainer)).toBeDefined();
  });
  it('render QuantityContainer', () => {
    const wrapper = shallow(<CheckoutItem {...mockProps} />);
    expect(wrapper.find(QuantityContainer)).toBeDefined();
  });
});
