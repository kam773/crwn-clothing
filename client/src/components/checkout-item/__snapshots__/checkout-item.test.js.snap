// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create shanpshot capture snapshot of CheckoutItem 1`] = `
<div
  className="sc-bdVaJa epBhhw"
>
  <div
    className="sc-bwzfXH hpBwHb"
  >
    <img
      alt="item"
      src="url"
    />
  </div>
  <span
    className="sc-htpNat HjiRX"
  >
    name
  </span>
  <span
    className="sc-htpNat sc-bxivhb jCdJTz"
  >
    <div
      onClick={[Function]}
    >
      ❮
    </div>
    <span>
      2
    </span>
    <div
      onClick={[Function]}
    >
      ❯
    </div>
  </span>
  <span
    className="sc-htpNat HjiRX"
  >
    20
  </span>
  <div
    className="sc-ifAKCX iRMGWo"
    onClick={[Function]}
  >
    ✕
  </div>
</div>
`;
